aten/src/ATen/native/cuda/Blas.cpp

```diff
@@ -181,6 +181,19 @@ Tensor& addmm_out_cuda_impl(Tensor& result, const Tensor& self, const Tensor& ma
   }
 
   if (&result != &self) {
+
+    // beta*self + alpha*(mat1@mat2) = 0
+    if ((mat1._is_zerotensor() || mat2._is_zerotensor() || alpha.equal(0.0)) &&
+        (self._is_zerotensor() || beta.equal(0.0))) {
+      result = at::_efficientzerotensor(self_sizes, self.options());
+      return result;
+    }
+
+    // beta*self
+    if (mat1._is_zerotensor() || mat2._is_zerotensor() || alpha.equal(0.0)){
+      return beta.equal(1) ? self.clone() : self * beta;
```

Comment:
```suggestion
      return self * beta;
```

since we are cloning anyway